das/s3_storage_service.go

```diff
@@ -110,7 +111,8 @@ func (s3s *S3StorageService) Put(ctx context.Context, value []byte, timeout uint
 		Bucket: aws.String(s3s.bucket),
 		Key:    aws.String(s3s.objectPrefix + EncodeStorageServiceKey(dastree.Hash(value))),
 		Body:   bytes.NewReader(value)}
-	if !s3s.discardAfterTimeout {
+	if !s3s.discardAfterTimeout && timeout <= math.MaxInt64 {
```

Comment:
Is this change on purpose?

Reply:
It's currently backwards, fixed in https://github.com/OffchainLabs/nitro/pull/2600